src/main.cpp

```diff
@@ -4850,61 +4862,68 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                 // Pruned nodes may have deleted the block, so check whether
                 // it's available before trying to send.
                 if (send && (mi->second->nStatus & BLOCK_HAVE_DATA))
-                {
+                  {
                     // Send block from disk
                     CBlock block;
                     if (!ReadBlockFromDisk(block, (*mi).second, consensusParams))
```

Comment:
This API clearly is unsound as it doesn't differentiate the element being absent from a catastrophic failure of the program (for instance, DB corruption). Patching around it seems like the wrong way to go about it.